tests/benchmark/bm_basics.cpp

```diff
@@ -0,0 +1,167 @@
+    std::ifstream input(file_name, std::ios::binary);
+    if (input.is_open()) {
+        serializer.loadIndex(file_name,
+                             reinterpret_cast<HNSWIndex *>(hnsw_index)->getSpace().get());
+    } else {
+        std::cerr << "HNSW index file was not found in path. Exiting..." << std::endl;
+        exit(1);
+    }
+    free(location);
+}
+
+class BM_VecSimBasics : public benchmark::Fixture {
+protected:
+    std::mt19937 rng;
+    VecSimIndex *bf_index;
+    VecSimIndex *hnsw_index;
+    size_t dim;
+    size_t n_vectors;
+    std::vector<float> query;
+
+    BM_VecSimBasics() {
+        dim = 100;
+        n_vectors = 1000000;
+        query.reserve(dim);
+    }
+
+public:
+    void SetUp(const ::benchmark::State &state) {
+        rng.seed(47);
+        std::vector<float> data(n_vectors * dim);
+        std::uniform_real_distribution<> distrib;
+        for (size_t i = 0; i < n_vectors * dim; ++i) {
+            data[i] = (float)distrib(rng);
+        }
+        VecSimParams params = {.algo = VecSimAlgo_BF,
+                               .bfParams = {.type = VecSimType_FLOAT32,
+                                            .dim = dim,
+                                            .metric = VecSimMetric_L2,
+                                            .initialCapacity = n_vectors}};
+        bf_index = VecSimIndex_New(&params);
+
+        // Add random vectors to Flat index.
+        for (size_t i = 0; i < n_vectors; ++i) {
+            VecSimIndex_AddVector(bf_index, data.data() + dim * i, i);
+        }
+        // Initialize and load HNSW index.
+        size_t M = 50;
+        size_t ef_c = 350;
+        size_t ef_r = 500;
+        params = {.algo = VecSimAlgo_HNSWLIB,
+                  .hnswParams = {.type = VecSimType_FLOAT32,
+                                 .dim = dim,
+                                 .metric = VecSimMetric_L2,
+                                 .initialCapacity = n_vectors + 1,
+                                 .M = M,
+                                 .efConstruction = ef_c,
+                                 .efRuntime = ef_r}};
+        hnsw_index = VecSimIndex_New(&params);
+
+        // Load pre-generated HNSW index containing the same vectors as the Flat index.
+        GetHNSWIndex(hnsw_index);
+
+        // Generate random query vector before test.
+        for (size_t i = 0; i < dim; ++i) {
+            query[i] = (float)distrib(rng);
+        }
+    }
+
+    void TearDown(const ::benchmark::State &state) {
+        VecSimIndex_Free(hnsw_index);
+        VecSimIndex_Free(bf_index);
+    }
+
+    ~BM_VecSimBasics() {}
+};
+
+BENCHMARK_DEFINE_F(BM_VecSimBasics, AddVectorHNSW)(benchmark::State &st) {
+
+    // Add a single vector to the index.
+    for (auto _ : st) {
+        VecSimIndex_AddVector(hnsw_index, query.data(), n_vectors);
+        st.PauseTiming();
+        VecSimIndex_DeleteVector(hnsw_index, n_vectors);
+        st.ResumeTiming();
```

Comment:
please explain that you pause the timing to avoid the in-place update. Also, `n_vectors` is confusing for the vector id